docs/DeveloperGuide.md

```diff
@@ -25,14 +29,213 @@
+- Please reference from the above Command Summary Reference.
+- Try to see your list of records by entering command `list`
+
+8. Initial launch
+    1. Test case: Launch the application for the first time <br>
+       Expected: Welcome message "Welcome to BookBob, Dr. Bob!" is displayed. A new `data` directory and data files are created if they do not yet exist.
+
+### Shutdown
+1. Enter `exit` to exit BookBob. 
+2. BookBob automatically saves your patient record data to a file named "bookbob_data.txt" in a "data" folder in the same directory as the Bookbob.jar file.
+
+3. Exit application
+    1. Test Case: `exit` <br>
+       Expected: Application terminates, all data are saved.
+
+## Patient Record Management
+
+1. Adding a patient
+    1. (Positive) Test Case (all fields): `add n/Mary Jane ic/S9876543A p/91234567 d/Fever m/Paracetamol ha/NUS PGPR dob/01011990 v/21-10-2024 15:48 al/Peanuts s/Female mh/Asthma` <br>
+       Expected: Patient Mary Jane is added with all provided information.
+
+    2. (Negative) Test Case (missing mandatory fields): `add n/Mary Jane p/91234567` <br>
+       Expected: Error message requesting NRIC and visit date.
+
+    3. (Negative) Test Case (invalid date format): `add n/Mary Jane ic/S9876543A v/2024-10-21` <br>
+       Expected output: Error message about invalid date format.
+
+2. Listing patients
+    1. (Positive) Test Case: `list` <br>
+       Expected: Displays all patient records with their details and visit history.
+
+    2. (Negative) Test Case (empty records): `list` when no patients exist <br>
+       Expected: "No patients found." message.
+
+3. Finding patients
+    1. (Positive) Test Case (by NRIC): `find ic/S1234567Z` <br>
+       Expected: Lists all patients whose NRIC contains "S1234567Z".
+
+    2. (Positive) Test Case (by name): `find n/John` <br>
+       Expected: Lists all patients whose names contain "John".
+
+    3. (Negative) Test Case (invalid search key): `find x/John` <br>
+       Expected: Error message about invalid search key.
+
+4. Editing patient records
+    1. (Positive) Test Case: `edit ic/S9876543A /to n/John Smith p/92345678` <br>
+       Expected: Patient's name updated to "John Smith" and phone number to "92345678".
+
+    2. (Negative) Test Case (non-existent patient): `edit ic/S0000000X /to n/John Smith` <br>
+       Expected: "No patient found." message.
+
+5. Deleting patient records
+    1. (Positive) Test Case: `delete S9876543A` <br>
+       Expected: Patient record is deleted and confirmation message shown.
+
+    2. (Negative) Test Case (non-existent NRIC): `delete S0000000X` <br>
+       Expected: Error message that patient not found.
+
+### Visit Management
+
+1. Adding visits
+    1. (Positive) Test Case: `addVisit ic/S9876543A v/21-10-2024 15:48 d/Fever,Cough m/Paracetamol,Cough Syrup` <br>
+       Expected: Visit added to patient's record with diagnoses and medications.
```

Comment:
Would providing a few additional examples for variations help reduce input errors?